Sources/Commands/SwiftPackageCollectionsTool.swift

```diff
@@ -0,0 +1,358 @@
+        abstract: "Interact with package collections",
+        discussion: "SEE ALSO: swift build, swift package, swift run, swift test",
+        version: Versioning.currentVersion.completeDisplayString,
+        subcommands: [
+            Add.self,
+            Describe.self,
+            DescribeCollection.self,
+            List.self,
+            ProfileList.self,
+            Refresh.self,
+            Remove.self,
+            Search.self
+        ],
+        helpNames: [.short, .long, .customLong("help", withSingleDash: true)])
+
+    public init() {
+    }
+
+    // MARK: Profiles
+
+    struct ProfileList: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "List configured profiles")
+
+        @OptionGroup
+        var jsonOptions: JSONOptions
+
+        mutating func run() throws {
+            let profiles: [PackageCollectionsModel.Profile] = try await { self.collections.listProfiles(callback: $0) }
+
+            if jsonOptions.json {
+                try JSONEncoder().print(profiles)
+            } else {
+                profiles.forEach {
+                    print($0)
+                }
+            }
+        }
+    }
+
+    // MARK: Collections
+
+    struct List: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "List configured collections")
+
+        @OptionGroup
+        var jsonOptions: JSONOptions
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        mutating func run() throws {
+            let collections = try await { self.collections.listCollections(identifiers: nil, in: profileOptions.usedProfile, callback: $0) }
+
+            if jsonOptions.json {
+                try JSONEncoder().print(collections)
+            } else {
+                collections.forEach {
+                    print("\($0.name) - \($0.source.url)")
+                }
+            }
+        }
+    }
+
+    struct Refresh: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "Refresh configured collections")
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        mutating func run() throws {
+            let collections = try await { self.collections.refreshCollections(in: profileOptions.usedProfile, callback: $0) }
+            print("Refreshed \(collections.count) configured package collections.")
+        }
+    }
+
+    struct Add: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "Add a new collection")
+
+        @Argument(help: "URL of the collection to add")
+        var collectionUrl: String?
+
+        @Option(name: .long, help: "Sort order for the added collection")
+        var order: Int?
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        mutating func run() throws {
+            guard let collectionUrlString = collectionUrl, let collectionUrl = URL(string: collectionUrlString) else {
```

Comment:
Nit: it might be nice to distinguish the "missing URL" vs "malformed URL" cases.

Reply:
I have solved this by making all arguments non-optional since argument-parser already takes care of error handling.